internal/report/report.go

```diff
@@ -1237,9 +1239,12 @@ func computeTotal(prof *profile.Profile, value, meanDiv func(v []int64) int64) i
 			diffDiv += d
 		}
 	}
+
 	if diffTotal > 0 {
 		total = diffTotal
 		div = diffDiv
+	} else if negSamples {
+		ui.PrintErr("Negative sample values appeared in the profile.\nIf using the -base flag to compare profiles, consider using the -diff_base flag instead.")
```

Comment:
It's a bit strange that printing out this user interface messages goes down as deeply as to this code that computes the total. As well as that report.New has to take the UI now. I think it would be better to make computeTotal return a boolean (in addition to the int64 like now) indicating whether negative values were seen, store that in the report object, and print the message in Generate() making that function take UI.

Reply:
Done.